src/IO/HDF5/HDF5Auxiliary.cpp

```diff
@@ -98,7 +98,7 @@ hid_t openPMD::GetH5DataType::operator()(Attribute const &att)
         return H5Tcopy(H5T_NATIVE_DOUBLE);
     case DT::LONG_DOUBLE:
     case DT::VEC_LONG_DOUBLE:
-        return H5Tcopy(H5T_NATIVE_LDOUBLE);
+        return H5Tcopy(m_userTypes.at(typeid(long double).name()));
```

Comment:
```suggestion
        return H5Tcopy(H5T_NATIVE_LDOUBLE);
```
I think that the change can be reverted again. In writing procedures, we can always use the native datatype anyway. In `HDF5IOHandlerImpl::readDataset()` we need to manually check if the dataset can be read as native `long double` anyway before attempting the workaround with the manually defined `long double`. As a result, it's fine to return native `long double` here in `::openDataset()`. Other reading procedures don't use this anyway.